test/libsolidity/Imports.cpp

```diff
@@ -172,6 +172,17 @@ BOOST_AUTO_TEST_CASE(filename_with_period)
 	BOOST_CHECK(!c.compile());
 }
 
+BOOST_AUTO_TEST_CASE(context_dependent_remappings_ensure_default_and_module_preserved)
+{
+	CompilerStack c;
+	c.setRemappings(vector<string>{"foo=vendor/foo_2.0.0", "vendor/bar:foo=vendor/foo_1.0.0", "bar=vendor/bar"});
+	c.addSource("main.sol", "import \"foo/foo.sol\"; import {Bar} from \"bar/bar.sol\"; contract Main is Foo2, Bar {} pragma solidity >=0.0;");
+	c.addSource("vendor/bar/bar.sol", "import \"foo/foo.sol\"; contract Bar {Foo1 foo;} pragma solidity >=0.0;");
+	c.addSource("vendor/foo_1.0.0/foo.sol", "contract Foo1 {} pragma solidity >=0.0;");
+	c.addSource("vendor/foo_2.0.0/foo.sol", "contract Foo2 {} pragma solidity >=0.0;");
+	BOOST_CHECK(c.compile());
+}
+
```

Comment:
Please add a test case for the situation I mentioned in https://github.com/ethereum/solidity/pull/1548#pullrequestreview-16164779 for both orders of remappings.

It is very important that the remappings do not depend on the order (except for the case where we have multiple remappings with identical context and prefix).